sample-schemas/demo_schema.erl

```diff
@@ -0,0 +1,60 @@
+-module(demo_schema).
+
+-include_lib("typerefl/include/types.hrl").
+
+-type ip() :: ipv4() | ipv6().
+-type ipv4() :: {0..255, 0..255, 0..255, 0..255}.
+-type ipv6() :: {0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535}.
+
+-reflect_type([ip/0]).
+
+-export([keys/0]).
+
+keys() ->
+    [ {"foo.setting", fun setting/1}
+    , {"foo.endpoint", fun endpoint/1}
+    , {"foo.greet", fun greet/1}
+    ].
```

Comment:
we may need to support some built in types like:

`{"foo.flag", bool}` which get translated into `fun hocon_schema_builtin:bool/1`.
same for ip type

Reply:
`hocon_schema_builtin:bool` would need to return a mapping name, which is impossible.

Reply:
instead we can omit `converter`. If `converter` is undefined, hocon_schema will look for default converter based on its type